chain/chain/src/runtime/mod.rs

```diff
@@ -882,7 +883,12 @@ impl RuntimeAdapter for NightshadeRuntime {
         if ProtocolFeature::StatelessValidation.enabled(next_protocol_version)
             || cfg!(feature = "shadow_chunk_validation")
         {
-            trie = trie.recording_reads_new_recorder();
+            let epoch_id =
+                self.epoch_manager.get_epoch_id_from_prev_block(&block.prev_block_hash)?;
```

Comment:
nit: I think you can get the epoch id from the block header, it may be a tad shorter

Reply:
Neither `ApplyChunkShardContext` nor `ApplyChunkBlockContext` have the epoch id 😞